### PR TITLE
impr: consider numbers in pb

### PR DESCRIPTION
**Pull request by notreallystatic**
**Fixes**:
https://github.com/monkeytypegame/monkeytype/issues/4823

### CODE REVIEW

**Comment by monkeytypegeorge** [comment]
Continuous integration check(s) failed. Please review the failing check's logs and make the necessary changes. https://github.com/monkeytypegame/monkeytype/actions/runs/7578828078

**Comment by Miodec** [comment]
Updating the types is not enough. You also need to handle this new property correctly when saving pbs, and make sure to display it on the frontend.

**Comment by notreallystatic** [comment]
@Miodec  Yeah, the pbs will be saved. Also, in UI it will be shown as a new personal best cause we will set the flag as true in response. Are there any other places in the UI, where we are showing the score explicitly that I need to handle? What else I am missing?

**Comment by Miodec** [comment]
> @Miodec Yeah, the pbs will be saved. Also, in UI it will be shown as a new personal best cause we will set the flag as true in response. Are there any other places in the UI, where we are showing the score explicitly that I need to handle? What else I am missing?

`pb-tables-popup.ts`

image

File: frontend/src/ts/db.ts [hunk 1]
```diff
@@ -636,6 +637,7 @@ export async function saveLocalPB<M extends MonkeyTypes.Mode>(
     ).forEach((pb) => {
       if (
         pb.punctuation === punctuation &&
+        !!pb.numbers === numbers &&
```
Comment (Miodec) [inline frontend/src/ts/db.ts:640]: I dont think `!!` is necessary here

Comment (notreallystatic) [reply frontend/src/ts/db.ts:640]: I checked that the older personal-bests in a given mode won't have the numbers field. That's why I added this.  Btw, I am also adding support for pace-caret and tags, so still working on that. Could you tell if I am missing anything else as well?

Comment (Miodec) [reply frontend/src/ts/db.ts:640]: Ah yeah, youre right. All though maybe i would prefer `(pb.numbers ?? false) === number`

**Pull request merged**